compiler/src/java_plugin/cpp/java_generator.cpp

```diff
@@ -785,18 +803,21 @@ static void PrintStub(
               "    getChannel().newCall($method_method_name$(), getCallOptions()), request);\n");
           break;
       }
+    } else {
+      GRPC_CODEGEN_FAIL << "Do not create Stub interfaces";
     }
     p->Outdent();
     p->Print("}\n");
   }
 
-  if (impl_base) {
+  if (impl_base && !interface) {
     p->Print("\n");
     p->Print(
         *vars,
         "@$Override$ public final $ServerServiceDefinition$ bindService() {\n");
-    (*vars)["instance"] = "this";
-    PrintBindServiceMethodBody(service, vars, p);
+    p->Indent();
```

Comment:
Just make this all literal in a single string.

Reply:
Done